zerver/lib/actions.py

```diff
@@ -1870,18 +1870,21 @@ def notify_reaction_update(user_profile: UserProfile, message: Message,
 
 def do_add_reaction(user_profile: UserProfile, message: Message,
                     emoji_name: str, emoji_code: str, reaction_type: str) -> None:
+    """Should be called while holding a select_for_update lock
+    (e.g. via access_message(..., lock_message=True)) on the
+    Message row, to prevent IntegrityErrors if the Message is deleted
+    by another thread while this is running.
```

Comment:
I'd still phrase this more generally e.g. `.... to prevent race conditions.`. 

"if the Message is deleted by another thread while this is running." is too specific, this isn't the only case we're defending against, e.g. the lock should protect from double-adding a reaction and we're removing the block that was handling that case due to it.

As a sidenote, "thread" would also not be the best term here - we don't really use multithreading much - these races generally happen between two parallel processes, not threads - that's because parallel http requests get handled by the server in parallel processes.

Reply:
How about this?
```
Should be called while holding a FOR UPDATE lock
(e.g. via access_message(..., lock_message=True)) on the
Message row, to prevent IntegrityErrors due to race conditions.

```